packages/job-manager/src/runner/runner.spec.ts

```diff
@@ -0,0 +1,195 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import 'reflect-metadata';
+
+import { Message, Queue } from 'azure-services';
+import { ServiceConfiguration } from 'common';
+import { Logger } from 'logger';
+import { IMock, It, Mock, Times } from 'typemoq';
+import { Batch } from '../batch/batch';
+import { JobTask, JobTaskState } from '../batch/job-task';
+import { PoolMetrics, PoolMetricsState } from '../batch/pool-metrics';
+import { Runner } from './runner';
+
+// tslint:disable: no-unsafe-any
+
+let runner: Runner;
+let batchMock: IMock<Batch>;
+let queueMock: IMock<Queue>;
+let poolMetricsMock: IMock<PoolMetrics>;
+let serviceConfigMock: IMock<ServiceConfiguration>;
+let loggerMock: IMock<Logger>;
+const targetQueuedTasksOverloadRatioDefault = 2;
+const tasksIncrementIntervalInSecondsDefault = 1;
+
+describe(Runner, () => {
+    beforeEach(() => {
+        process.env.AZ_BATCH_JOB_ID = 'batch-job-id';
+        batchMock = Mock.ofType(Batch);
+        queueMock = Mock.ofType(Queue);
+        poolMetricsMock = Mock.ofType(PoolMetrics);
+        serviceConfigMock = Mock.ofType(ServiceConfiguration);
+        loggerMock = Mock.ofType(Logger);
+
+        queueMock.setup(o => o.scanQueue).returns(() => 'scan-queue');
+
+        serviceConfigMock
+            .setup(async s => s.getConfigValue('jobManagerConfig'))
+            .returns(async () => {
+                return {
+                    targetQueuedTasksOverloadRatio: targetQueuedTasksOverloadRatioDefault,
+                    tasksIncrementIntervalInSeconds: tasksIncrementIntervalInSecondsDefault,
+                    periodicRestartInHours: 1,
+                };
+            })
+            .verifiable(Times.atLeastOnce());
+
+        batchMock
+            .setup(async o => o.createJobIfNotExists(process.env.AZ_BATCH_JOB_ID, true))
+            .returns(async () => Promise.resolve(process.env.AZ_BATCH_JOB_ID))
+            .verifiable(Times.once());
+
+        runner = new Runner(batchMock.object, queueMock.object, poolMetricsMock.object, serviceConfigMock.object, loggerMock.object);
+        runner.runOnce = true;
```

Comment:
>   runner.runOnce = true; [](start = 5, length = 25)

you should also have test that to validate that you run this in a loop.

Reply:
can you give an example?